lib/pub_grub/version_solver.rb

```diff
@@ -109,8 +109,8 @@ def next_package_to_try
       solution.unsatisfied.min_by do |term|
         package = term.package
         range = term.constraint.range
-        matching_versions = source.versions_for(package, range)
-        higher_versions = source.versions_for(package, range.upper_invert)
+        matching_versions = source.sorted_versions_for(package, range)
+        higher_versions = source.sorted_versions_for(package, range.upper_invert)
```

Comment:
I guess this would be a breaking change since Source wasn't previously required to define this method 🤔

I see that Gel overrides this method with a Solver subclass anyways, maybe I should just propose that kind of change directly to Bundler?